content/general/events/aws-lambda.textile

```diff
@@ -0,0 +1,163 @@
+---
+title: AWS Lambda Functions
+section: general
+index: 22
+jump_to:
+  Help with:
+    - Fields#fields
+    - Restrictions#restrictions
+    - Example#example
+---
+
+As part of "Reactor Events":/general/events, it is possible to integrate with various systems.
+
+"AWS Lambda Functions":https://aws.amazon.com/lambda provide event-driven serverless compute functions which allow users to easily run code whenever events are sent to it. These can be very useful for integrating into various other AWS systems.
+
+h3(#tutorials). Tutorials & Examples
+
+If you would like to just dive in and see an example of this being implemented, then take a look at our "AWS Lambda tutorial":/tutorials/reactor-event-aws.
+
+h2(#fields). Rule fields
+
+- AWS Region := The region you chose for your "AWS Lambda function":https://docs.aws.amazon.com/general/latest/gr/rande.html#lambda_region
+- Function Name := The name you gave your "AWS Lambda function":https://docs.aws.amazon.com/lambda/latest/dg/getting-started.html
+- AWS Authentication Scheme := the "authentication scheme":https://support.ably.io/support/solutions/articles/3000071908 you wish to use for your function. Either @AWS Credentials@ or @ARN of an assumable role@
+- Qualifier := _optional_ qualifier for your AWS Lambda function
+- "Source":/general/events#sources := Choose which of @Message@ or @Presence@ events on channels should activate this Reactor Event Rule.
+- Channel filter := An optional filter, which allows the Rule to be applied to a restricted set of channels. This can be specified as a regular expression, allowing for swathes of channels to be used
+- Encoding := The encoding to be used by this Rule. This can only be JSON for AWS Lambda Functions
+
+If the Rule is using *AWS Credentials* to authenticate, it will also have the following options:
+
+- AWS Credentials := a key:value pair which match your "AWS credentials":https://support.ably.io/support/solutions/articles/3000071908
+
+If the Rule is using *ARN of an assumable role* to authenticate, it will also have the following options:
+
+- Assume Role ARN := your "ARN":https://support.ably.io/support/solutions/articles/3000071908 which will work for this AWS Lambda function
+
+h2(#restrictions). Restrictions
+
+At present, it is not possible to "batch":/general/event#batching messages to AWS Lambda Functions, nor can messages be "encoded":/general/events#encoding in anything besides JSON.
+
+h2(#example). Examples
+
+h3(#envelope-examples). Enveloped event payloads
+
+Enveloped messages do not contain any additional headers beyond the "shared headers":/general/events#examples.
```

Comment:
lambda function rules have no concept of headers. We invoke the function with a payload, that's it